channeldb/channel.go

```diff
@@ -531,6 +531,29 @@ func (c *OpenChannel) ChanStatus() ChannelStatus {
 	return c.chanStatus
 }
 
+// ApplyChanStatus allows the caller to modify the internal channel state in a
+// thead-safe manner.
+func (c *OpenChannel) ApplyChanStatus(status ChannelStatus) {
+	c.Lock()
+	defer c.Unlock()
+
+	c.chanStatus |= status
```

Comment:
this will make it out of sync with the db version. Should instead use `putChanStatus`?